detox/src/devices/runtime/drivers/ios/SimulatorDriver.js

```diff
@@ -63,11 +63,16 @@ class SimulatorDriver extends IosDriver {
     }
   }
 
-  async installApp(binaryPath) {
+  async resetAppState(binaryPath, bundleId) {
+    await this._uninstallApp(bundleId);
+    await this._installApp(binaryPath);
+  }
+
+  async _installApp(binaryPath) {
     await this._applesimutils.install(this.udid, getAbsoluteBinaryPath(binaryPath));
   }
 
-  async uninstallApp(bundleId) {
+  async _uninstallApp(bundleId) {
```

Comment:
You can't really do that. `installApp()` and `uninstallApp()` are public methods the driver must override...